File: src/main/java/com/github/bjoernpetersen/jmusicbot/InitStateWriter.java
package com.github.bjoernpetersen.jmusicbot;

import javax.annotation.Nonnull;

public interface InitStateWriter {

  @Nonnull
  InitStateWriter NO_OP = new InitStateWriter() {
    @Override
    public void begin(@Nonnull Plugin plugin) {
    }

    @Override
    public void state(@Nonnull String state) {
    }

    @Override
    public void warning(@Nonnull String warning) {
    }
  };

  void begin(@Nonnull Plugin plugin);

  void state(@Nonnull String state);

  void warning(@Nonnull String warning);

  default void close() {
  }
}

File: src/main/kotlin/com/github/bjoernpetersen/musicbot/spi/plugin/management/InitStateWriter.kt
package com.github.bjoernpetersen.musicbot.spi.plugin.management

import com.github.bjoernpetersen.musicbot.spi.plugin.Plugin

interface InitStateWriter {
    fun begin(plugin: Plugin)
    fun state(state: String)
    fun warning(warning: String)
    fun close() {}
}
